src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTestForSearchQuery.java

```diff
@@ -0,0 +1,139 @@
+    private static final String RAS_SEARCH_QUERY_ROLE_ASSIGNMENT_URL = "/am/role-assignments/query";
+
+    @BeforeEach
+    public void setUpEachTest() throws InterruptedException {
+        Thread.sleep(2000);
+    }
+
+    @After
+    void teardown() {
+        Executor.closeIdleConnections();
+    }
+
+    @Pact(provider = "am_role_assignment_service_search_query", consumer = "am_org_role_mapping")
+    public RequestResponsePact executeSearchQueryRoleAssignmentAndGet200(PactDslWithProvider builder)
+            throws IOException {
+
+        return builder
+                .given("A list of role assignments for the search query")
+                .uponReceiving("RAS takes s2s/auth token and returns search query results")
+                .path(RAS_SEARCH_QUERY_ROLE_ASSIGNMENT_URL)
+                .method(HttpMethod.POST.toString())
+                .body(createRoleAssignmentRequestSearchQuery(), String.valueOf(ContentType.JSON))
+                .willRespondWith()
+                .status(HttpStatus.OK.value())
+                .headers(getResponseHeaders())
+                .body(createRoleAssignmentResponseSearchQuery())
+                .toPact();
+    }
+
+    private String createRoleAssignmentRequestSearchQuery() {
+        return "{\n"
+                + "\"actorId\": [\"234873\"]\n"
+                + "}";
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "executeSearchQueryRoleAssignmentAndGet200")
+    void getSearchQueryResultsByActorIdAndGet200Test(MockServer mockServer)
+            throws JSONException, IOException {
+        String actualResponseBody =
+                SerenityRest
+                        .given()
+                        .headers(getHttpHeaders())
+                        .contentType(ContentType.JSON)
+                        .body(createRoleAssignmentRequestSearchQuery())
+                        .post(mockServer.getUrl() + RAS_SEARCH_QUERY_ROLE_ASSIGNMENT_URL)
+                        .then()
+                        .log().all().extract().asString();
+
+        JSONObject jsonResponse = new JSONObject(actualResponseBody);
+        JSONArray roleAssignmentResponse = (JSONArray)jsonResponse.get("roleAssignmentResponse");
+        JSONObject first = (JSONObject)roleAssignmentResponse.get(0);
+        assertThat(first.get("actorId"), equalTo(ACTOR_ID));
+    }
+
+    private DslPart createRoleAssignmentResponseSearchQuery() {
+        return newJsonBody(o -> o
+            .minArrayLike("links", 1, 1, link -> link.stringType("rel", "binary")
+                    .stringType("href", "http://localhost:4096/am/role-assignments/query"))
+            .minArrayLike("roleAssignmentResponse", 1, 1,
+                    roleAssignmentResponse -> roleAssignmentResponse
+                            .stringType("id", "14a21569-eb80-4681-b62c-6ae2ed069e6f")
+                            .stringValue("actorIdType", "IDAM")
+                            .stringValue("actorId", ACTOR_ID)
```

Comment:
matching the actor id can cause issue. It should be based upon type but check with provider.